app/models/miq_set_membership.rb

```diff
@@ -0,0 +1,4 @@
+class MiqSetMembership < ApplicationRecord
+  # NOTE:  Can't set `belongs_to :miq_set` here because `MiqSet` is not a model
```

Comment:
It is, and it is why I am not going the route of converting this to full blown STI inheritance, because `Tag` and tagging was big barrier.

This is a note to fix this in the future, but the current implementation is to get this across the finish line and make forward progress with the least risk possible (though there still is some)

Reply:
See https://github.com/ManageIQ/manageiq/pull/21227#issuecomment-848106329 for reference